src/main/java/co/cask/gcp/datastore/source/util/DatastoreSourceSchemaUtil.java

```diff
@@ -0,0 +1,185 @@
+
+  private static final Logger LOG = LoggerFactory.getLogger(DatastoreSourceSchemaUtil.class);
+
+  /**
+   * Validates if key alias column is present in the schema and its type is {@link Schema.Type#STRING}.
+   *
+   * @param schema CDAP schema
+   * @param keyAlias key alias
+   */
+  public static void validateKeyAlias(Schema schema, String keyAlias) {
+    Schema.Field field = schema.getField(keyAlias);
+    if (field == null) {
+      throw new IllegalArgumentException(String.format("Key field [%s] must exist in the schema.", keyAlias));
+    }
+
+    Schema fieldSchema = field.getSchema();
+    Schema.Type type = fieldSchema.getType();
+    if (Schema.Type.STRING != type) {
+      String foundType = fieldSchema.isNullable()
+        ? "nullable " + fieldSchema.getNonNullable().getType().name()
+        : type.name();
+      throw new IllegalArgumentException(String.format("Key field [%s] type must be non-nullable STRING, not %s",
+          keyAlias, foundType));
+    }
+  }
+
+  /**
+   * Validates given CDAP schema types to be compliant with Datastore existing types.
+   *
+   * @param schema CDAP schema
+   */
+  public static void validateSchema(Schema schema) {
+    List<Schema.Field> fields = schema.getFields();
+    if (fields == null || fields.isEmpty()) {
+      throw new IllegalArgumentException("Source schema must contain at least one field");
+    }
+    fields.stream()
+      .map(Schema.Field::getSchema)
+      .forEach(DatastoreSourceSchemaUtil::validateFieldSchema);
+  }
+
+  /**
+   * Constructs CDAP schema based on given CDAP entity and source configuration,
+   * will add Datastore key to the list of schema fields if config include key flag is set to true.
+   *
+   * @param entity Datastore entity
+   * @param config Datastore source config
+   * @return CDAP schema
+   */
+  public static Schema constructSchema(Entity entity, DatastoreSourceConfig config) {
+    List<Schema.Field> fields = constructSchemaFields(entity);
+
+    if (config.isIncludeKey()) {
+      fields.add(Schema.Field.of(config.getKeyAlias(), Schema.of(Schema.Type.STRING)));
+    }
+
+    return Schema.recordOf("schema", fields);
+  }
+
+  /**
+   * Validates given field schema to be complaint with Datastore types.
+   * Will throw {@link IllegalArgumentException} if schema contains unsupported type.
+   *
+   * @param fieldSchema schema for CDAP field
+   */
+  private static void validateFieldSchema(Schema fieldSchema) {
+    switch (fieldSchema.getType()) {
+      case STRING:
+      case DOUBLE:
+      case BOOLEAN:
+      case BYTES:
+        return;
+      case LONG:
+        // timestamps in CDAP are represented as LONG with TIMESTAMP_MICROS logical type
+        Schema.LogicalType logicalType = fieldSchema.getLogicalType();
+        if (logicalType == null || Schema.LogicalType.TIMESTAMP_MICROS == logicalType) {
+          return;
+        }
+        throw new IllegalArgumentException("Unsupported logical type for Long: " + logicalType);
```

Comment:
handle logical types in a separate switch statement. Also include the field name in the error message